src/app/queries/types/vms.types.ts

```diff
@@ -26,12 +32,50 @@ export interface IVMwareVM {
   cpuCount: number;
   coresPerSocket: number;
   memoryMB: number;
-  guestName: string;
   balloonedMemory: number;
   ipAddress: string;
   networks: IVMwareObjRef[];
   disks: IVMwareVMDisk[];
-  concerns: IVMwareVMConcern[];
+  concerns: ISourceVMConcern[];
   revisionValidated: number;
   isTemplate: boolean;
 }
+
+export interface IRHVNIC {
+  id: string;
+  name: string;
+  interface: string;
+  profile: {
+    id: string;
+    name: string;
+    description: string;
+    selfLink: string;
+    revision: number;
+    network: string;
+    qos: string;
+  };
+}
+
+export interface IRHVDiskAttachment {
+  id: string;
+  disk: {
+    id: string;
+    name: string;
+    description: string;
+    selfLink: string;
+    revision: number;
+    shared: boolean;
+    profile: string;
+    storageDomain: string;
+  };
+}
+
+export interface IRHVVM extends IBaseSourceVM {
+  description: string;
+  cluster: string;
+  host: string;
+  nics: IRHVNIC[];
+  diskAttachments: IRHVDiskAttachment[];
+}
+
+export type SourceVM = IVMwareVM | IRHVVM;
```

Comment:
To follow our naming convention, please rename to "ISourceVM"

Reply:
I've been using the I prefix only for interfaces, there are a few other places where I have types that are unions of multiple interfaces and don't use this prefix. I think the distinction is useful because for example, you can't extend this type (I don't think). In the future maybe we could use `TSourceVM` or something to indicate that it is a type.

Reply:
@mturley, you're right, no need for a type. Eyes getting tired and mixed up with `IBaseSourceVM`.